src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyTextFilter.java

```diff
@@ -0,0 +1,620 @@
+     * matches {@code nearbyTextPattern}.
+     */
+    @XdocsPropertyType(PropertyType.PATTERN)
+    private String checkPattern = DEFAULT_CHECK_PATTERN;
+
+    /** Specify check violation message pattern to suppress. */
+    @XdocsPropertyType(PropertyType.PATTERN)
+    private String messagePattern;
+
+    /** Specify check ID pattern to suppress. */
+    @XdocsPropertyType(PropertyType.PATTERN)
+    private String idPattern;
+
+    /**
+     * Specify negative/zero/positive value that defines the number of lines
+     * preceding/at/following the suppressing nearby text. Property can also be a RegExp group
+     * index at {@code nearbyTextPattern} in format of {@code $x} and be picked
+     * from line that matches {@code nearbyTextPattern}.
+     */
+    private String lineRange = DEFAULT_LINE_RANGE;
+
+    /** The absolute path to the currently processed file. */
+    private String currentFileAbsolutePath = "";
+
+    /**
+     * Setter to specify nearby text pattern to trigger filter to begin suppression.
+     *
+     * @param pattern a {@code Pattern} value.
+     */
+    public final void setNearbyTextPattern(Pattern pattern) {
+        nearbyTextPattern = pattern;
+    }
+
+    /**
+     * Setter to specify check name pattern to suppress. Property can also
+     * be a RegExp group index at {@code nearbyTextPattern} in
+     * format of {@code $x} and be picked from line that matches {@code nearbyTextPattern}.
+     *
+     * @param pattern a {@code String} value.
+     */
+    public final void setCheckPattern(String pattern) {
+        checkPattern = pattern;
+    }
+
+    /**
+     * Setter to specify check violation message pattern to suppress.
+     *
+     * @param pattern a {@code String} value.
+     */
+    public void setMessagePattern(String pattern) {
+        messagePattern = pattern;
+    }
+
+    /**
+     * Setter to specify check ID pattern to suppress.
+     *
+     * @param pattern a {@code String} value.
+     */
+    public void setIdPattern(String pattern) {
+        idPattern = pattern;
+    }
+
+    /**
+     * Setter to specify negative/zero/positive value that defines the number
+     * of lines preceding/at/following the suppressing nearby text. Property can also
+     * be a RegExp group index at {@code nearbyTextPattern} in
+     * format of {@code $x} and be picked from line that matches {@code nearbyTextPattern}.
+     *
+     * @param format a {@code String} value.
+     */
+    public final void setLineRange(String format) {
+        lineRange = format;
+    }
+
+    @Override
+    public boolean accept(AuditEvent event) {
+        boolean accepted = true;
+
+        if (event.getViolation() != null) {
+            final String eventFileTextAbsolutePath = event.getFileName();
+
+            if (!currentFileAbsolutePath.equals(eventFileTextAbsolutePath)) {
+                final FileText currentFileText = getFileText(eventFileTextAbsolutePath);
```

Comment:
looks like caching ....
rename currentFileAbsolutePath -> cachedFileAbsolutePath

as there should be "current" in stateless Filters, ideally filters should be stateless, to be ready for Multithread mode of execution (future).

as we agree that it is caching, we might use some different model later on.

Reply:
Done.